src/carnot/funcs/builtins/collections_test.cc

```diff
@@ -103,7 +153,6 @@ TEST(CollectionsTest, CanSerializeDeserialize_Float) {
   }
 
   auto any_uda = AnyUDA<types::Float64Value>();
-  EXPECT_NE(uda_tester.Result(), any_uda.Finalize(nullptr));
```

Comment:
In response to this comment, we updated this PR to `EXPECT_DEATH` in lieu of the prior expectations. In the original impl., the `AnyUDA` would yield a value, even if it was uninitialized -- essentially, it would provide whatever value was default constructed for the UDA data type. In this PR, we added a `DCHECK` that requires the UDA be initialized before providing a value. This `DCHECK` will fail if we use the original test expectations, hence we change those original test expectations to `EXPECT_DEATH`.